ui/pages/confirmations/confirm-transaction-base/confirm-transaction-base.component.js

```diff
@@ -268,6 +273,11 @@ export default class ConfirmTransactionBase extends Component {
     ) {
       this.updateValueToMax();
     }
+
+    if (hasPriorityApprovalRequest && !prevHasPriorityApprovalRequest) {
+      // Redirect to home which will redirect to the priority confirmation.
+      history.push(mostRecentOverviewPage);
+    }
```

Comment:
Can you stop the routing to this component in the first place for PriorityApprovalRequest so that we do not have to route back.

Reply:
Unfortunately not, as the scenario in this bug is when you have two transactions queued, then accept one.

At this point, there is a small delay while the background prepares and submits the smart transaction, before creating the approval request that we can then redirect to.

But the spinner is shown throughout so it's not too jarring for the user.